File: app/src/main/java/org/codepond/imdemo/service/chat/MessagingService.java
package org.codepond.imdemo.service.chat;

import org.codepond.imdemo.ChatMessage;

public interface MessagingService {
    interface OnMessageReceivedListener {
        void onMessageReceived(ChatMessage chatMessage);
    }

    void sendMessage(ChatMessage chatMessage);
    void setCurrentParticipant(String participantJid);
    void setOnMessageReceivedListener(OnMessageReceivedListener listener);
}

File: app/src/main/java/org/codepond/imdemo/service/chat/MessagingService.kt
package org.codepond.imdemo.service.chat

import org.codepond.imdemo.model.ChatMessage

interface MessagingService {
    interface OnMessageReceivedListener {
        fun onMessageReceived(chatMessage: ChatMessage)
    }

    fun sendMessage(chatMessage: ChatMessage)
    fun setCurrentParticipant(participantJid: String)
    fun setOnMessageReceivedListener(listener: OnMessageReceivedListener?)
}
